spec/solver_spec.rb

```diff
@@ -16,7 +18,13 @@
       expect { @solver.factorial(-1) }.to raise_error(RuntimeError, 'Argument given is negative!')
     end
 
-    # You can write the TESTS for reverse function
+    it 'factorial should return 0 when input is 1' do
```

Comment:
- [ ] Kindly check your factorial function , the test is failing:
![image](https://user-images.githubusercontent.com/61408860/188083644-2fec4820-92b3-4f7e-9a87-e7d499f6833e.png)